ledger-core/application/benchs/single_pulse_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package benchs
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/neilotoole/errgroup"
+	"github.com/paulbellamy/ratecounter"
+
+	"github.com/insolar/assured-ledger/ledger-core/application/testutils/launchnet"
+	"github.com/insolar/assured-ledger/ledger-core/instrumentation/inslogger/instestlogger"
+)
+
+func BenchmarkSinglePulse(b *testing.B) {
+	instestlogger.SetTestOutput(b)
+
+	for numNodes := 2; numNodes <= 5; numNodes++ {
+		b.Run(fmt.Sprintf("Nodes %d", numNodes), func(b *testing.B) {
+
+			res := launchnet.CustomRunWithoutPulsar(numNodes, 0, 0, func(apiAddresses []string) int {
```

Comment:
Can you extract that function to separate named function? maximum nesting became 6.